docs/mobile-apps/automated-testing/appium/appium-flutter.md

```diff
@@ -0,0 +1,625 @@
+
+5. The `import` from the previous step provides an `enableFlutterDriverExtension()` function that enables the Flutter Driver extension. Add the following code statement in the main.dart file of your Flutter project to enable it for your project.
+  ```dart
+  void main() {
+    // This line enables the extension
+    enableFlutterDriverExtension();
+
+    runApp(const MyApp());
+  }
+  ```
+  Check out the [`enableFlutterDriverExtension`](https://api.flutter.dev/flutter/flutter_driver_extension/enableFlutterDriverExtension.html) function section to learn more about it's powers.
+
+## Building your Flutter App
+After you have enabled the Flutter Driver extension, you can now build your app for testing. To do so, follow the steps below
+
+1. Open your Flutter project in your favorite IDE and or open a terminal in the root directory of your Flutter project.
+2. Determine for which device type (Android Emulator/Real Device or iOS Simulator/Real Device) you want to build your app for.
+3. Run one of the following commands
+
+  | OS | Device Type | Build Command | Output Folder |
+  | --- | --- | --- | --- |
+  | Android | Emulator | `flutter build apk --debug` | `build/app/outputs/flutter-apk` |
+  | Android | Real Device | `flutter build apk --debug` | `build/app/outputs/flutter-apk`|
+  | iOS | Simulator | `flutter build ios --debug` | `build/ios/iphoneos` |
+  | iOS | Real Device | `flutter build ipa --profile` | `build/ios/ipa` |
+
+## Uploading your Flutter App to Sauce Labs
+
+You now need to manually upload the built apps to the preferred Data Center, see [Manually Uploading an App](/mobile-apps/live-testing/live-mobile-app-testing/#uploading-an-app) for instructions.
+
+
+## Configuring your Appium Capabilities
+More information on how to write Appium tests for Flutter apps can be found in the [Appium Flutter Driver](https://github.com/appium-userland/appium-flutter-driver/) repository. You can also find a sample Flutter app and tests in the [Demo JS - Appium Flutter](https://github.com/saucelabs-training/demo-js/tree/docs-1.2/webdriverio/appium-app/examples/appium-flutter)-repository.
+
+:::note W3C Capabilities
+We encourage you to use W3C capabilities for your tests. For more information on W3C capabilities, see [W3C Capabilities](/dev/test-configuration-options/#mobile-appium-capabilities).
+:::
+
+Apps that have been build with Flutter 2 can use Appium 1 and Appium 2, while apps that have been build with Flutter 3 can only use Appium 2.
+
+:::note
+The Virtual Device Cloud with Android Emulators and iOS Simulators currently only supports Flutter Apps that have been build with Flutter 2. If you need to test your Flutter 3 Apps we encourage you to use our Real Device Cloud. For the corresponding capabilities, see [Real Device Capabilities](#real-devices).
+:::
+
+### Android Emulators and iOS Simulators
```

Comment:
The tabs here are a little awkward. It might be better to separate this into two sections, one with configs for Android Emulators and one with the configs for iOS Simulators.

Reply:
I think I would have put them in two sections under different headings. But I just noticed that we use this style in the Migrating to Appium 2.0 as well. So, since we have a precedent, you can leave this as is.